### PR TITLE
Enable yamllint

### PR DESCRIPTION
**Pull request by ssbarnea**
- fixed config bug which prevented custom rule from being loaded
- adds yamllint to pre-commit so we run it

While this will fail to lint it will uncover how inconsistent is our current yaml formatting.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -34,6 +34,13 @@ repos:
   - id: check-merge-conflict
   - id: debug-statements
     language_version: python3
+- repo: https://github.com/adrienverge/yamllint.git
```
Comment (webknjaz) [inline .pre-commit-config.yaml:37]: Oh, I didn't know that we don't run it...

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -34,6 +34,13 @@ repos:
...
+  rev: v1.23.0
+  hooks:
+  - id: yamllint
+    files: \.(yaml|yml)$
+    types: [file, yaml]
```
Comment (webknjaz) [inline .pre-commit-config.yaml:42]: I'm pretty sure this should be
```suggestion
    types: [yaml]
```

File: .azure-pipelines/tox.yml [hunk 3]
```diff
@@ -4,36 +4,36 @@ parameters:
...
 jobs:
-- template: run-tox-env.yml@tox
-  parameters:
-    jobs:
-      docs: null
-      lint:
-        before:
-        - task: CacheBeta@0
-          displayName: cache pre-commit
+  - template: run-tox-env.yml@tox
```
Comment (webknjaz) [inline .azure-pipelines/tox.yml:7]: Sequences shouldn't be indented.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (ssbarnea) [reply .pre-commit-config.yaml:42]: See https://github.com/adrienverge/yamllint/blob/master/.pre-commit-hooks.yaml -- that is the recommended setup. Keep in mind that these work with AND and not OR.

File: .azure-pipelines/tox.yml (hunk 3 above)
Comment (ssbarnea) [reply .azure-pipelines/tox.yml:7]: For Ansible code in particular that is a serious issue, especially with blocks where it makes extremely hard to find block/rescue/always sections. 

Also non identing makes is much easier to misplace a when condition between block level or last task within the block.

For these reasons all the Ansible code used by OpenStack does enforce sequence-indentation, something this change does **not**. This change is more relaxed, only request to be consistent inside the file.

So basically this touched only the files that reported inconsistencies, not others.
